Comments/Export/ExportToExcel.cs

```diff
@@ -390,17 +391,21 @@ private async Task<(List<Excel> collatedData, bool showOrganisationExpressionOfI
 		{
 			List<Excel> excel = new List<Excel>();
 
-			var userIds = comments.Select(comment => comment.CreatedByUserId).Concat(answers.Select(answer => answer.CreatedByUserId)).Distinct();
-			var currentUserDetails = _userService.GetCurrentUserDetails();
+			IEnumerable<string> userIdsForComments = comments.Select(comment => comment.SubmissionComment?.FirstOrDefault()?.Submission.SubmissionByUserId);
+			IEnumerable<string> userIdsForAnswers = answers.Select(answer => answer.SubmissionAnswer?.FirstOrDefault()?.Submission.SubmissionByUserId);
+			IEnumerable<string> allUserIds = userIdsForComments.Concat(userIdsForAnswers);
+			IEnumerable<string> uniqueUserIds = allUserIds.Where(id => id != null).Distinct();
```

Comment:
this might be better factored out to a little function, as only the uniqueUserIds are used. the other variables are irrelevant to the rest of this method.